### PR TITLE
Update documentation

### PR DESCRIPTION
**Pull request by dweindl**
* Update documentation
* Remove obsolete parts
* Fix typos, formatting, ...

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,4 +1,4 @@
-<img src="https://raw.githubusercontent.com/AMICI-dev/AMICI/master/documentation/gfx/banner.png" height="60" align="left">
+<img src="https://raw.githubusercontent.com/AMICI-dev/AMICI/master/documentation/gfx/banner.png" height="60" align="left" alt="AMICI logo">
```
Comment (yannikschaelte) [inline README.md:1]: :+1:

File: README.md [hunk 2]
```diff
@@ -12,8 +12,8 @@ AMICI provides a multi-language (Python, C++, Matlab) interface for the
 (for algebraic differential equations). AMICI allows the user to read
 differential equation models specified as [SBML](http://sbml.org/)
 or [PySB](http://pysb.org/)
-and automatically compiles such models into `.mex` simulation files
-(Matlab), C++ executables or Python modules.
+and automatically compiles such models into Python modules, C++ libraries or
+`.mex` simulation files (Matlab).
```
Comment (yannikschaelte) [inline README.md:16]:
```suggestion
Matlab `.mex` simulation files.
```
more consistent with the others ...

File: python/amici/custom_commands.py [hunk 3]
```diff
@@ -319,7 +320,7 @@ def set_compiler_specific_extension_options(
     for attr in ['extra_compile_args', 'extra_link_args']:
         try:
             new_value = getattr(ext, attr) + \
-                getattr(ext, f'{attr}_{compiler_type}')
+                        getattr(ext, f'{attr}_{compiler_type}')
```
Comment (yannikschaelte) [inline python/amici/custom_commands.py:323]: flake8 would not agree

File: python/amici/gradient_check.py [hunk 4]
```diff
@@ -163,7 +162,7 @@ def check_derivatives(model: Model,
         whether to check least squares related values.
 
     :param skip_zero_pars:
-        whether to perform FD checks for paramameters that are zero
+        whether to perform FD checks for parameters that are zero
```
Comment (yannikschaelte) [inline python/amici/gradient_check.py:165]: not paramamameters?

File: python/amici/pysb_import.py [hunk 5]
```diff
@@ -262,8 +263,9 @@ def _process_pysb_expressions(pysb_model: pysb.Model,
         pysb model
 
     :param observables:
-        list of names of pysb.Expressions or pysb.Observables that are to be
-        mapped to ODEModel observables
+        list of names of :class`pysb.Expression`\\ s or
+        :class:`pysb.Observable` \\s that are to be mapped to ODEModel
```
Comment (yannikschaelte) [inline python/amici/pysb_import.py:267]:
```suggestion
        :class:`pysb.Observable`\\ s that are to be mapped to ODEModel
```

File: python/amici/pysb_import.py [hunk 6]
```diff
@@ -1114,9 +1116,9 @@ def has_fixed_parameter_ic(specie: pysb.core.ComplexPattern,
         ODE model
 
     :return:
-        False if the species does not have an initial condition at all.
+        ``false`` if the species does not have an initial condition at all.
```
Comment (yannikschaelte) [inline python/amici/pysb_import.py:1119]:
```suggestion
        ``False`` if the species does not have an initial condition at all.
```

File: python/amici/sbml_import.py [hunk 7]
```diff
@@ -1385,12 +1385,12 @@ def _reduce_stoichiometry(self, species_solver, volume_updates) -> List:
...
 
         :param volume_updates:
-            List with updates for the stoichiometrix matrix accounting for
+            List with updates for the stoichiometric matrix accounting for
             compartment volumes
 
         :returns volume_updates_solver:
             List (according to reduced stoichiometry) with updates for the
-            stoichiometrix matrix accounting for compartment volumes
+            stoichiometric matrix accounting for compartment volumes
```
Comment (yannikschaelte) [inline python/amici/sbml_import.py:1393]: what was up there???

**Review by yannikschaelte — APPROVED** [review]

File: python/amici/sbml_import.py (hunk 7 above)
Comment (dweindl) [reply python/amici/sbml_import.py:1393]: Can only guess that somebody was watching Asterix while coding...

File: python/amici/pysb_import.py (hunk 6 above)
Comment (dweindl) [reply python/amici/pysb_import.py:1119]: Although I don't like sentences starting in lower-case, I am not sure if one should change case of symbols.

File: python/amici/custom_commands.py (hunk 3 above)
Comment (dweindl) [reply python/amici/custom_commands.py:323]: Mine actually does

File: python/amici/pysb_import.py (hunk 6 above)
Comment (yannikschaelte) [reply python/amici/pysb_import.py:1119]: False is upper case in python ... or am I messing up with R here ...

Comment (dweindl) [reply python/amici/pysb_import.py:1119]: No... I'm messing up with C++

File: include/amici/solver.h [hunk 8]
```diff
@@ -1501,7 +1501,7 @@ class Solver {
     void applyQuadTolerances() const;
 
     /**
-     * @brief updates all senstivivity solver tolerances according to the
+     * @brief updates all sensitivity solver tolerances according to the
```
Comment (yannikschaelte) [inline include/amici/solver.h:1504]: Good to know it's _not_ spelled that way.

**Pull request merged**